src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,86 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int COFFICIENT_GROW = 2;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_LOAD_FACTOR * DEFAULT_CAPACITY);
+    }
+
+    private static class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        private Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(key, value, null);
+        int index = (key == null) ? 0 : Math.abs(key.hashCode() % table.length);
+        Node<K, V> nodeByIndex = table[index];
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            while (nodeByIndex.next != null || Objects.equals(nodeByIndex.key, key)) {
```

Comment:
Will it work without it  (Objects.equals(nodeByIndex.key, key)) ?

Reply:
no) becouse we can have null key

Reply:
Ця умова дублює умову всередині циклу, її потрібно прибрати
```suggestion
            while (nodeByIndex.next != null) {
```

Reply:
змінив